packages/kbn-es-query/src/filters/build_filters/types.ts

```diff
@@ -14,6 +14,18 @@ import { PhraseFilter } from './phrase_filter';
 import { RangeFilter } from './range_filter';
 import { MatchAllFilter } from './match_all_filter';
 import { MissingFilter } from './missing_filter';
+import { DslQuery } from '../../kuery';
+
+/**
+ * Serializable state is something is a POJO JavaScript object that can be
+ * serialized to a JSON string.
+ */
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type SerializableState = {
+  [key: string]: Serializable;
+};
+export type SerializableValue = string | number | boolean | null | undefined | SerializableState;
```

Comment:
```suggestion
export interface SerializableArray extends Array<SerializableValue> {}
export type SerializableValue = string | number | boolean | null | undefined | SerializableState | SerializableArray;

```

You need to have `SerializableValue` also include `SerializableArray`.
Then all casting to `SerializableState` becomes unnecessary, as it will be compatible with `JsonObject`.